clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/spark/datawriter/jobs/AbstractDataWriterSparkJob.java

```diff
@@ -0,0 +1,461 @@
+          pushJobSetting.sourceKafkaInputVersionInfo.getCompressionStrategy().name());
+      jobConf.set(
+          KAFKA_INPUT_SOURCE_TOPIC_CHUNKING_ENABLED,
+          pushJobSetting.sourceKafkaInputVersionInfo.isChunkingEnabled());
+    } else {
+      jobConf.set(VALUE_SCHEMA_ID_PROP, pushJobSetting.valueSchemaId);
+      jobConf.set(DERIVED_SCHEMA_ID_PROP, pushJobSetting.derivedSchemaId);
+    }
+    jobConf.set(ENABLE_WRITE_COMPUTE, pushJobSetting.enableWriteCompute);
+
+    if (!props.containsKey(KAFKA_PRODUCER_REQUEST_TIMEOUT_MS)) {
+      // If the push job plug-in doesn't specify the request timeout config, default will be infinite
+      jobConf.set(KAFKA_PRODUCER_REQUEST_TIMEOUT_MS, Integer.MAX_VALUE);
+    }
+    if (!props.containsKey(KAFKA_PRODUCER_RETRIES_CONFIG)) {
+      // If the push job plug-in doesn't specify the retries config, default will be infinite
+      jobConf.set(KAFKA_PRODUCER_RETRIES_CONFIG, Integer.MAX_VALUE);
+    }
+    if (!props.containsKey(KAFKA_PRODUCER_DELIVERY_TIMEOUT_MS)) {
+      // If the push job plug-in doesn't specify the delivery timeout config, default will be infinite
+      jobConf.set(KAFKA_PRODUCER_DELIVERY_TIMEOUT_MS, Integer.MAX_VALUE);
+    }
+
+    jobConf.set(TELEMETRY_MESSAGE_INTERVAL, props.getString(TELEMETRY_MESSAGE_INTERVAL, "10000"));
+    jobConf.set(EXTENDED_SCHEMA_VALIDITY_CHECK_ENABLED, pushJobSetting.extendedSchemaValidityCheckEnabled);
+
+    // Compression related
+    // Note that COMPRESSION_STRATEGY is from topic creation response as it might be different from the store config
+    // (eg: for inc push)
+    jobConf.set(
+        COMPRESSION_STRATEGY,
+        pushJobSetting.topicCompressionStrategy != null
+            ? pushJobSetting.topicCompressionStrategy.name()
+            : CompressionStrategy.NO_OP.name());
+    jobConf.set(
+        ZSTD_COMPRESSION_LEVEL,
+        props.getString(ZSTD_COMPRESSION_LEVEL, String.valueOf(Zstd.maxCompressionLevel())));
+    jobConf.set(ZSTD_DICTIONARY_CREATION_SUCCESS, pushJobSetting.isZstdDictCreationSuccess);
+
+    /**
+     * Pass-through the properties whose names start with:
+     * <ul>
+     *   <li> {@link VeniceWriter.VENICE_WRITER_CONFIG_PREFIX} </li>
+     *   <li> {@link ApacheKafkaProducerConfig.KAFKA_CONFIG_PREFIX} </li>
+     *   <li> {@link KafkaInputRecordReader.KIF_RECORD_READER_KAFKA_CONFIG_PREFIX} </li>
+     * </ul>
+     *
+     * Override the properties that are specified with the {@link SPARK_DATA_WRITER_CONF_PREFIX} prefix.
+     **/
+    List<String> passThroughPrefixList = Arrays.asList(
+        VeniceWriter.VENICE_WRITER_CONFIG_PREFIX,
+        KAFKA_CONFIG_PREFIX,
+        KafkaInputRecordReader.KIF_RECORD_READER_KAFKA_CONFIG_PREFIX);
+    int passThroughPrefixListSize = passThroughPrefixList.size();
+    /**
+     * The following logic will make sure there is no prefix that is a prefix of another prefix.
+     */
+    for (int i = 0; i < passThroughPrefixListSize; ++i) {
```

Comment:
I didn't want to touch it since it was basically a copy of the MR code, but I think I'll just move it to a test case instead of having it here. I don't see a point in having this checked at runtime